barretenberg/cpp/src/barretenberg/vm2/simulation/bitwise.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <cstdint>
+
+#include "barretenberg/vm2/common/memory_types.hpp"
+#include "barretenberg/vm2/simulation/context.hpp"
+#include "barretenberg/vm2/simulation/events/bitwise_event.hpp"
+#include "barretenberg/vm2/simulation/events/event_emitter.hpp"
+#include "barretenberg/vm2/simulation/memory.hpp"
+
+namespace bb::avm2::simulation {
+
+class BitwiseInterface {
+  public:
+    virtual ~BitwiseInterface() = default;
+    virtual void and_op(MemoryTag tag, uint128_t a, uint128_t b, uint128_t c) = 0;
```

Comment:
One more thing, uint128_t might need to be taken by `const uint128_t&`. It looks like a primitive type (like uint64_t) etc. But I think it's defined in BB and it's actually a class/struct?